src/mail/view/MailViewerViewModel.ts

```diff
@@ -478,8 +497,12 @@ export class MailViewerViewModel {
 
 	/** @return list of inline referenced cid */
 	private async loadMailBody(mail: Mail): Promise<string[]> {
-		// Short-circuit if we already loaded everything. Will avoid resetting contentBlockingStatus or mail body.
-		if (this.sanitizeResult) {
+		if (
+			this.renderedMail != null && haveSameId(mail, this.renderedMail)
+			&& mail.state !== MailState.DRAFT
```

Comment:
what is the purpose of singling out drafts here?

Reply:
we don't want to exit early for drafts here because they may actually have been edited, whereas non-drafts can't be edited so we don't need to reload them if we have already loaded them. I'll improve the comment